### ISSUE TITLE
add `mask` option to stats

### ISSUE THREAD

**Issue by petrelharp** [tskit-dev/tskit#346]
This is going to be necessary for stdpopsim. We can do it easily in python at first as I [did here](https://github.com/petrelharp/treestats_ms/blob/c706c88a882646fa5d4a627a970101cdaf961646/figures/plot_1kg_diversity.py#L88).

**Comment by jeromekelleher** [comment]
Agreed. As discussed somewhere in #261, I think the ``mask`` argument should be a 2D numpy array with shape (N, 2) so that we would mask out two regions with something like

```python
ts.diversity(mask=[(s1, e1), (s2, e2)])
```
We can use the recently added interval operations in util.py to parse these mask arguments.

**Comment by jeromekelleher** [comment]
As discussed, we're not sure when these are needed though. I'm provisionally adding them to a new 0.2.3 milestone, but we can promote to 0.2.2 if we want.

**Comment by szhan** [comment]
Is this feature still desired?

**Comment by jeromekelleher** [comment]
Yes - but it's a can of worms. I wouldn't take it on unless you have a spare month!
